src/lib.rs

```diff
@@ -78,70 +82,65 @@ impl Default for AttestationAgent {
 }
 
 impl AttestationAgent {
+    /// Create a new instance of [AttestationAgent].
     pub fn new() -> Self {
-        let kbc_module_list = KbcModuleList::new();
-        let kbc_instance_map = HashMap::new();
         AttestationAgent {
-            kbc_module_list,
-            kbc_instance_map,
+            kbc_module_list: KbcModuleList::new(),
+            kbc_instance_map: HashMap::new(),
         }
     }
 
     fn register_instance(&mut self, kbc_name: String, kbc_instance: KbcInstance) {
         self.kbc_instance_map.insert(kbc_name, kbc_instance);
     }
 
-    fn instantiate_kbc(&mut self, kbc_name: String, kbs_uri: String) -> Result<()> {
-        let instantiate_func = self.kbc_module_list.get_func(&kbc_name)?;
-        let kbc_instance = (instantiate_func)(kbs_uri);
-        self.register_instance(kbc_name, kbc_instance);
+    fn instantiate_kbc(&mut self, kbc_name: &str, kbs_uri: &str) -> Result<()> {
+        let instantiate_func = self.kbc_module_list.get_func(kbc_name)?;
+        let kbc_instance = (instantiate_func)(kbs_uri.to_string());
+        self.register_instance(kbc_name.to_string(), kbc_instance);
         Ok(())
     }
 
     #[allow(dead_code)]
     fn check(&self, kbc_name: String) -> Result<KbcCheckInfo> {
-        let kbc_instance = self
-            .kbc_instance_map
+        self.kbc_instance_map
             .get(&kbc_name)
-            .ok_or_else(|| anyhow!("The KBC instance does not existing!"))?;
-        let check_info: KbcCheckInfo = kbc_instance.check()?;
-        Ok(check_info)
+            .ok_or_else(|| anyhow!("The KBC instance does not existing!"))?
+            .check()
     }
 }
 
 #[async_trait]
 impl AttestationAPIs for AttestationAgent {
     async fn decrypt_image_layer_annotation(
         &mut self,
-        kbc_name: String,
-        kbs_uri: String,
-        annotation: String,
+        kbc_name: &str,
+        kbs_uri: &str,
+        annotation: &str,
     ) -> Result<Vec<u8>> {
-        if self.kbc_instance_map.get_mut(&kbc_name).is_none() {
-            self.instantiate_kbc(kbc_name.clone(), kbs_uri)?;
+        if !self.kbc_instance_map.contains_key(kbc_name) {
+            self.instantiate_kbc(kbc_name, kbs_uri)?;
         }
-        let kbc_instance = self
-            .kbc_instance_map
-            .get_mut(&kbc_name)
-            .ok_or_else(|| anyhow!("The KBC instance does not existing!"))?;
-        let plain_payload = kbc_instance.decrypt_payload(&annotation).await?;
-        Ok(plain_payload)
+        self.kbc_instance_map
+            .get_mut(kbc_name)
+            .ok_or_else(|| anyhow!("The KBC instance does not existing!"))?
```

Comment:
While we're here can we change `existing` -> `exist`.

Reply:
fixed